### DISCUSSION TITLE
web3auth.authenticateUser() metamask mobile

### DISCUSSION THREAD

**Discussion by jpshankle** [Web3Auth/Web3Auth#534] [Help]
Is there a way to get web3auth.authenticateUser() to open metamask mobile automatically? It appears that it will not switch to the metamask mobile app automatically to let the user sign?

I've triggered web3auth.authenticateUser() automatically and used click event without async, but it looks in the code it's async so it may not work as stated in this doc https://docs.walletconnect.com/mobile-linking.

Anybody have any good workarounds if this is the case?

**Comment by shahbaz17** [comment]
Hey @jpshankle 

You're using `web3auth.authenticateUser()` on web. Right? And want to open metamask. Can you share some information about mobile, os and SDK versions?

Also, what does it open upon using this function?

**Reply by jpshankle** [reply]
Using it on mobile web on iOS latest web3auth base 2.0.0 web3auth 2.0.2.

**Reply by jpshankle** [reply]
It doesn't open anything, it does create the sign in metamask mobile but you have to manually switch applications

**Reply by jpshankle** [reply]
This is a known issue with wallet connect. Before I switched to web3auth i was successfully doing this with siwe bc i just did the sign action in the correct order. It would be helpful if there could be a compatible version of authenticateUser

**Reply by shahbaz17** [reply]
Hey @jpshankle 

`web3auth.authenticateUser()` is currently available on Web. Will share more info here once this comes to Mobile SDKs.
